packages/jsii-dotnet-generator/src/Amazon.JSII.Generator/TypeProxyGeneratorBase.cs

```diff
@@ -0,0 +1,101 @@
+            : base(package, type, symbols, namespaces)
+        {
+        }
+
+        protected override MemberDeclarationSyntax CreateType()
+        {
+            return SF.ClassDeclaration
+            (
+                CreateAttributes(),
+                SF.TokenList(SF.Token(SyntaxKind.InternalKeyword)),
+                GetProxyTypeNameSyntax(),
+                null,
+                CreateBaseList(),
+                SF.List<TypeParameterConstraintClauseSyntax>(),
+                SF.List(CreateMembers())
+            );
+
+            SyntaxList<AttributeListSyntax> CreateAttributes()
+            {
+                var typeOfExpression = SF.TypeOfExpression(Symbols.GetNameSyntax(Type));
+                var fullyQualifiedNameLiteral = SF.Literal(Type.FullyQualifiedName);
+
+                return SF.List(new[]
+                {
+                    SF.AttributeList(SF.SeparatedList(new[]
+                    {
+                        SF.Attribute(
+                            SF.ParseName("JsiiInterfaceProxy"),
+                            SF.ParseAttributeArgumentList($"({typeOfExpression}, {fullyQualifiedNameLiteral})")
+                        )
+                    }))
+                });
+            }
+
+            BaseListSyntax CreateBaseList()
+            {
+                return SF.BaseList(SF.SeparatedList(GetBaseTypes()));
+
+                IEnumerable<BaseTypeSyntax> GetBaseTypes()
+                {
+                    if (IsImplementingInterface)
+                    {
+                        yield return SF.SimpleBaseType(SF.ParseTypeName("DeputyBase"));
+                    }
+
+                    Namespaces.Add(Type);
+                    yield return SF.SimpleBaseType(Symbols.GetNameSyntax(Type, disambiguate: true));
+                }
+            }
+
+            IEnumerable<MemberDeclarationSyntax> CreateMembers()
+            {
+                return CreateConstructors()
+                    .Concat(CreateProperties())
+                    .Concat(CreateMethods());
+            }
+        }
+
+        protected virtual IEnumerable<MemberDeclarationSyntax> CreateConstructors()
+        {
+            yield return SF.ConstructorDeclaration
+            (
+                SF.List<AttributeListSyntax>(),
+
+                // Only Amazon.JSII.Runtime should create interface proxies,
+                // so we make the constructor private.
+                SF.TokenList(SF.Token(SyntaxKind.PrivateKeyword)),
+                GetProxyTypeNameSyntax(),
+                SF.ParseParameterList("(ByRefValue reference)"),
+                SF.ConstructorInitializer
+                (
+                    SyntaxKind.BaseConstructorInitializer,
+                    SF.ParseArgumentList("(reference)")
+                ),
+                SF.Block(),
+                null
+            );
+        }
+
+        private bool IsImplementingInterface => Type is InterfaceType;
```

Comment:
Nit: Property name is confusing, and no need to add a property for this. Just inline `if (Type is InterfaceType)` in `CreateType.CreateBaseList`.

Reply:
I was following the style some other classes were using. I can make this change for now though.